VSharp.SILI/Types.fs

```diff
@@ -57,6 +65,18 @@ module public Types =
     let private numericTypes = new HashSet<System.Type>(Seq.append integerTypes realTypes)
 
     let private primitiveTypes = new HashSet<Type>(Seq.append numericTypes [typedefof<bool>; typedefof<string>])
+        
+    let rec public IsAssignableToGenericType (givenType : Type) (genericType : Type) =
+        let isFindInterfaces =
+            givenType.GetInterfaces() |> 
```

Comment:
givenType.GetInterfaces() |> Array.exists (fun it -> it.IsGenericTypeDefinition && it.GetGenericTypeDefinition() = genericType)